bool product1
---
bool success